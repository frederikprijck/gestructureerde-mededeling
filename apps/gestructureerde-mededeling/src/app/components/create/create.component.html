<ogm-number [ogm]="ogm$ | async"></ogm-number>

<ogm-input
  (ogmInputChange)="ogmInputChangeHandler($event)"
  placeholderMessage="vul een getal in van maximaal 10 cijfers"
></ogm-input>
