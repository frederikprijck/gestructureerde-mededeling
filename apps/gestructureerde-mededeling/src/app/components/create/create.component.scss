@import '../../../assets/styles/screen-media-queries';

.ogm-form {
  text-align: center;
  padding: 0 10%;
}

input {
  text-align: center;
  width: 100%;
  margin: 1em 0;
  padding: 8px 24px;
  border-radius: 4px;
  border: 1px solid #e9e9e9;
  font-size: 2.5rem;
  box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14),
    0px 1px 3px 0px rgba(0, 0, 0, 0.12);
  outline: none;
  color: #103183;

  &::placeholder {
    color: #e1e1e1;
  }
}

.controls {
  margin: 2rem 0.5rem;
  display: flex;
  justify-content: space-around;

  button {
    font-size: 1.5rem;
    background-color: #18b351;

    &:active {
      background-color: #fff;
      color: #18b351;
      cursor: pointer;
    }
  }

  &.large {
    button {
      height: 60px;
      flex: 1 1 0;
      line-height: 60px;
      max-width: 450px;
      margin: 0 16px;

      text-overflow: ellipsis;
      white-space: nowrap;
      overflow: hidden;

      fa-icon {
        line-height: 1.5rem;
        padding-right: 12px;
      }

      font-size: 1.5rem;

      @media #{$mat-md} {
        max-width: 250px;
        height: 50px;
        font-size: 18px;
        line-height: 50px;

        mat-icon {
          padding-right: 0.5rem;
        }
      }

      @media #{$mat-lt-md} {
        margin: 0 0.5rem;
        height: 40px;
        font-size: 1rem;
        line-height: 40px;

        mat-icon {
          padding-right: 6px;
        }
      }
    }
  }
}

::ng-deep.mat-simple-snackbar {
  > span {
    width: 100%;
    text-align: center;
  }
}
