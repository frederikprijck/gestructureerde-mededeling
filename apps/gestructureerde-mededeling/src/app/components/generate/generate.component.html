<ogm-number [ogm]="ogm.numberFormat"></ogm-number>

<ng-container *ngIf="isSmallScreen$ | async; then smallControls; else largeControls"></ng-container>

<ng-template #smallControls>
  <div class="controls">
    <button mat-fab disableRipple color="primary" type="button" title="Verniewen" (click)="refresh()">
      <fa-icon [icon]="['fad', 'sync-alt']"></fa-icon>
    </button>
    <button
      class="ogm-copy"
      mat-fab
      disableRipple
      color="primary"
      type="button"
      title="Kopieer nummer"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm.number"
      (click)="copyToClipboard('Nummer gekopieerd naar klembord')"
    >
      <fa-icon [icon]="['fad', 'comment-dots']"></fa-icon>
    </button>
    <button
      class="ogm-copy"
      mat-fab
      disableRipple
      color="primary"
      type="button"
      title="Kopieer mededeling"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm.numberFormat"
      (click)="copyToClipboard('Mededeling gekopieerd naar klembord')"
    >
      <fa-icon [icon]="['fad', 'comment-lines']"></fa-icon>
    </button>
  </div>
</ng-template>

<ng-template #largeControls>
  <div class="controls large">
    <button mat-raised-button isRoundButton disableRipple color="primary" (click)="refresh()">
      <fa-icon [icon]="['fad', 'sync-alt']"></fa-icon>Vernieuwen
    </button>
    <button
      class="ogm-copy"
      mat-raised-button
      isRoundButton
      disableRipple
      color="primary"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm.number"
      (click)="copyToClipboard('Nummer gekopieerd naar klembord')"
    >
      <fa-icon [icon]="['fad', 'comment-dots']"></fa-icon>Kopieer nummer
    </button>
    <button
      class="ogm-copy"
      mat-raised-button
      isRoundButton
      disableRipple
      color="primary"
      data-clipboard-action="copy"
      [attr.data-clipboard-text]="ogm.numberFormat"
      (click)="copyToClipboard('Mededeling gekopieerd naar klembord')"
    >
      <fa-icon [icon]="['fad', 'comment-lines']"></fa-icon>Kopieer mededeling
    </button>
  </div>
</ng-template>
