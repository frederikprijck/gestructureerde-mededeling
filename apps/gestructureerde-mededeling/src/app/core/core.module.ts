import { NgModule } from '@angular/core';
import { RoutingModule } from './routing.module';
import { MaterialModule } from './material.module';
import { ReactiveFormsModule } from '@angular/forms';

@NgModule({
  declarations: [],
  imports: [RoutingModule, MaterialModule, ReactiveFormsModule],
  exports: [RoutingModule, MaterialModule, ReactiveFormsModule],
})
export class CoreModule {}
