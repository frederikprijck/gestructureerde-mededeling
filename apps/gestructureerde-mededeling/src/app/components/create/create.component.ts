import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { OgmService } from '../../core/services/ogm.service';
import { OgmInputChange } from '../../core/components/ogm-input/ogm-input.component';
import { ScreenService } from '../../core/services/screen.service';
import { MatSnackBar } from '@angular/material/snack-bar';
import { OgmData } from '../../core/ogm.model';
import * as ClipboardJS from 'clipboard';
import { map } from 'rxjs/operators';
import { Breakpoints } from '@angular/cdk/layout';
import { faCommentDots, faCommentLines, faSyncAlt } from '@fortawesome/pro-duotone-svg-icons';
import { FaIconLibrary } from '@fortawesome/angular-fontawesome';

@Component({
  selector: 'ogm-create',
  templateUrl: './create.component.html',
  styleUrls: ['./create.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CreateComponent implements OnDestroy {
  readonly isSmallScreen$: Observable<boolean>;
  readonly ogm$: BehaviorSubject<string>;

  ogm: OgmData;
  clipboard: ClipboardJS;

  constructor(
    private ogmService: OgmService,
    private screenService: ScreenService,
    private snackBar: MatSnackBar,
    private faIconLibrary: FaIconLibrary
  ) {
    this.isSmallScreen$ = this.screenService
      .observerBreakpoints()
      .pipe(map((breakpoints: { [key: string]: boolean }) => breakpoints[Breakpoints.XSmall]));

    this.ogm$ = new BehaviorSubject<string>(this.ogmService.init());

    this.clipboard = new ClipboardJS('.ogm-copy');

    this.faIconLibrary.addIcons(faSyncAlt, faCommentDots, faCommentLines);
  }

  ngOnDestroy() {
    this.clipboard.destroy();
  }

  ogmInputChangeHandler({ ogm }: OgmInputChange) {
    this.ogm = { number: this.ogmService.clean(ogm), numberFormat: ogm };

    this.ogm$.next(ogm);
  }

  copyToClipboard(message: string) {
    this.showSnackbarMessage(message);
  }

  showSnackbarMessage(message?: string) {
    if (message) {
      this.snackBar.open(message, '', {
        panelClass: 'center-clipboard-text',
        verticalPosition: 'top',
        duration: 750,
      });
    }
  }
}
