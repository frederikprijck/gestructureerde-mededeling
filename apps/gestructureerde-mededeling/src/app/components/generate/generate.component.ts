import { ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
import * as ClipboardJS from 'clipboard';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { Breakpoints } from '@angular/cdk/layout';
import { MatSnackBar } from '@angular/material/snack-bar';
import { ScreenService } from '../../core/services/screen.service';
import { OgmService } from '../../core/services/ogm.service';
import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
import { faSyncAlt, faCommentDots, faCommentLines } from '@fortawesome/pro-duotone-svg-icons';
import { OgmData } from '../../core/ogm.model';

@Component({
  selector: 'ogm-generate',
  templateUrl: './generate.component.html',
  styleUrls: ['./generate.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class GenerateComponent implements OnInit, OnDestroy {
  readonly isSmallScreen$: Observable<boolean>;

  ogm: OgmData;
  clipboard: ClipboardJS;

  constructor(
    private ogmService: OgmService,
    private screenService: ScreenService,
    private snackBar: MatSnackBar,
    private faIconLibrary: FaIconLibrary
  ) {
    this.isSmallScreen$ = this.screenService
      .observerBreakpoints()
      .pipe(map((breakpoints: { [key: string]: boolean }) => breakpoints[Breakpoints.XSmall]));

    this.clipboard = new ClipboardJS('.ogm-copy');

    this.faIconLibrary.addIcons(faSyncAlt, faCommentDots, faCommentLines);
  }

  ngOnInit() {
    this.refresh();
  }

  ngOnDestroy() {
    this.clipboard.destroy();
  }

  refresh(message?: string) {
    this.showSnackbarMessage(message);

    const ogmNumber = this.ogmService.generate();
    const ogmNumberFormat = this.ogmService.format(ogmNumber);

    this.ogm = { number: ogmNumber, numberFormat: ogmNumberFormat };
  }

  copyToClipboard(message: string) {
    this.showSnackbarMessage(message);
  }

  showSnackbarMessage(message?: string) {
    if (message) {
      this.snackBar.open(message, '', {
        panelClass: 'center-clipboard-text',
        verticalPosition: 'top',
        duration: 750,
      });
    }
  }
}
