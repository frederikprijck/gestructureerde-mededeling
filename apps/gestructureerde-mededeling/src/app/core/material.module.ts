import { NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { LayoutModule } from '@angular/cdk/layout';
import { MatIconModule } from '@angular/material/icon';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatToolbarModule } from '@angular/material/toolbar';

@NgModule({
  declarations: [],
  imports: [
    BrowserAnimationsModule,
    MatButtonModule,
    MatCardModule,
    MatIconModule,
    LayoutModule,
    MatSnackBarModule,
    MatToolbarModule,
  ],
  exports: [MatButtonModule, MatCardModule, MatIconModule, LayoutModule, MatSnackBarModule, MatToolbarModule],
})
export class MaterialModule {}
