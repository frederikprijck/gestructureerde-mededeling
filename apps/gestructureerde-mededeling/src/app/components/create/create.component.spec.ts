import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { CreateComponent } from './create.component';
import { OgmService } from '../../core/services/ogm.service';
import { Component, Input } from '@angular/core';
import { OgmInputChange } from '../../core/components/ogm-input/ogm-input.component';
import { MatSnackBar } from '@angular/material/snack-bar';
import { MaterialModule } from '../../core/material.module';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { GenerateComponent } from '../generate/generate.component';
import { ScreenService } from '../../core/services/screen.service';

@Component({
  selector: 'ogm-number',
  template: `
    <div>{{ ogm | json }}</div>
    <div>{{ isValid | json }}</div>
  `,
})
class MockNumberComponent {
  @Input() ogm: string;
  @Input() isValid: boolean;
}

@Component({
  selector: 'ogm-input',
  template: `
    <div>{{ validate | json }}</div>
    <div>{{ placeholderMessage | json }}</div>
  `,
})
class MockInputComponent {
  @Input() validate: boolean;
  @Input() placeholderMessage: string;
}

describe('CreateComponent', () => {
  let component: CreateComponent;
  let fixture: ComponentFixture<CreateComponent>;
  let snackBar: MatSnackBar;
  let ogmService: OgmService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [MaterialModule, FontAwesomeModule],
      declarations: [CreateComponent, MockNumberComponent, MockInputComponent],
      providers: [OgmService, ScreenService, MatSnackBar],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CreateComponent);
    component = fixture.componentInstance;

    snackBar = TestBed.inject(MatSnackBar);
    ogmService = TestBed.inject(OgmService);
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('rendering component', () => {
    it('should match snapshot', () => {
      // act
      fixture.detectChanges();

      // check
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('ogmInputChangeHandler', () => {
    it('should call next on ogm$ stream', () => {
      // prepare
      jest.spyOn(component.ogm$, 'next');

      const ogmInputChange: OgmInputChange = { ogm: '54321', isValid: undefined };

      // act
      component.ogmInputChangeHandler(ogmInputChange);

      // check
      expect(component.ogm$.next).toHaveBeenCalledWith('54321');
    });
  });
});
