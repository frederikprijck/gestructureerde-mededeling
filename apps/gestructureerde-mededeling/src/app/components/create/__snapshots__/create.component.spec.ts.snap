// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateComponent rendering component should match snapshot 1`] = `
<ogm-create
  clipboard={[Function Clipboard]}
  faIconLibrary={[Function FaIconLibrary]}
  isSmallScreen$={[Function Observable]}
  ogm$={[Function BehaviorSubject]}
  ogmService={[Function OgmService]}
  screenService={[Function ScreenService]}
  snackBar={[Function MatSnackBar]}
>
  <ogm-number>
    <div>
      "+++   /    /     +++"
    </div>
    <div>
      
    </div>
  </ogm-number><ogm-input
    placeholdermessage="vul een getal in van maximaal 10 cijfers"
  >
    <div>
      
    </div>
    <div>
      "vul een getal in van maximaal 10 cijfers"
    </div>
  </ogm-input><div
    class="controls large"
  >
    <button
      class="ogm-copy mat-focus-indicator mat-raised-button mat-button-base mat-primary"
      color="primary"
      data-clipboard-action="copy"
      disableripple=""
      isroundbutton=""
      mat-raised-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        <fa-icon
          class="ng-fa-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-comment-dots fa-w-16"
            data-icon="comment-dots"
            data-prefix="fad"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              class="fa-group"
            >
              <path
                class="fa-secondary"
                d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5a8 8 0 0 0-1.5 8.7A7.83 7.83 0 0 0 8 480c66.3 0 116-31.8 140.6-51.4A305 305 0 0 0 256 448c141.4 0 256-93.1 256-208S397.4 32 256 32zM128 272a32 32 0 1 1 32-32 32 32 0 0 1-32 32zm128 0a32 32 0 1 1 32-32 32 32 0 0 1-32 32zm128 0a32 32 0 1 1 32-32 32 32 0 0 1-32 32z"
                fill="currentColor"
              />
              <path
                class="fa-primary"
                d="M128 208a32 32 0 1 0 32 32 32 32 0 0 0-32-32zm128 0a32 32 0 1 0 32 32 32 32 0 0 0-32-32zm128 0a32 32 0 1 0 32 32 32 32 0 0 0-32-32z"
                fill="currentColor"
              />
            </g>
          </svg>
        </fa-icon>
        Kopieer nummer 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="ogm-copy mat-focus-indicator mat-raised-button mat-button-base mat-primary"
      color="primary"
      data-clipboard-action="copy"
      disableripple=""
      isroundbutton=""
      mat-raised-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        <fa-icon
          class="ng-fa-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-comment-lines fa-w-16"
            data-icon="comment-lines"
            data-prefix="fad"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              class="fa-group"
            >
              <path
                class="fa-secondary"
                d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5a8 8 0 0 0-1.5 8.7A7.83 7.83 0 0 0 8 480c66.3 0 116-31.8 140.6-51.4A305 305 0 0 0 256 448c141.4 0 256-93.1 256-208S397.4 32 256 32zm32 264a8 8 0 0 1-8 8H136a8 8 0 0 1-8-8v-16a8 8 0 0 1 8-8h144a8 8 0 0 1 8 8zm96-96a8 8 0 0 1-8 8H136a8 8 0 0 1-8-8v-16a8 8 0 0 1 8-8h240a8 8 0 0 1 8 8z"
                fill="currentColor"
              />
              <path
                class="fa-primary"
                d="M376 176H136a8 8 0 0 0-8 8v16a8 8 0 0 0 8 8h240a8 8 0 0 0 8-8v-16a8 8 0 0 0-8-8zm-96 96H136a8 8 0 0 0-8 8v16a8 8 0 0 0 8 8h144a8 8 0 0 0 8-8v-16a8 8 0 0 0-8-8z"
                fill="currentColor"
              />
            </g>
          </svg>
        </fa-icon>
        Kopieer mededeling 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </div>
</ogm-create>
`;
